cpp/include/proton/map.hpp

```diff
@@ -26,6 +26,8 @@
 #include "./internal/pn_unique_ptr.hpp"
 
 #include <cstddef>
+#include <initializer_list>
+#include <map>
```

Comment:
I'm not sure this is correct given my comment below - you may need pair though as that is used in the initializer_list constructor